set.mm

```diff
@@ -659185,6 +659196,173 @@ D Fn ( I ... ( M - 1 ) ) /\
+      ( cgrp wcel cfv co wceq grprinv syl2anc ) ADNOFBOFFEPCQGRLMBCDEFGHIJKST
+      $.
+  $}
+
+  ${
+    grpasscan2d.b $e |- B = ( Base ` G ) $.
+    grpasscan2d.p $e |- .+ = ( +g ` G ) $.
+    grpasscan2d.n $e |- N = ( invg ` G ) $.
+    grpasscan2d.g $e |- ( ph -> G e. Grp ) $.
+    grpasscan2d.1 $e |- ( ph -> X e. B ) $.
+    grpasscan2d.2 $e |- ( ph -> Y e. B ) $.
+    $( An associative cancellation law for groups.  (Contributed by SN,
+       29-Jan-2025.) $)
+    grpasscan2d $p |- ( ph -> ( ( X .+ ( N ` Y ) ) .+ Y ) = X ) $=
+      ( cgrp wcel cfv co wceq grpasscan2 syl3anc ) ADNOFBOGBOFGEPCQGCQFRKLMBCDE
+      FGHIJST $.
+  $}
+
+  ${
+    grpcominv.b $e |- B = ( Base ` G ) $.
+    grpcominv.p $e |- .+ = ( +g ` G ) $.
+    grpcominv.n $e |- N = ( invg ` G ) $.
+    grpcominv.g $e |- ( ph -> G e. Grp ) $.
+    grpcominv.x $e |- ( ph -> X e. B ) $.
+    grpcominv.y $e |- ( ph -> Y e. B ) $.
+    grpcominv.1 $e |- ( ph -> ( X .+ Y ) = ( Y .+ X ) ) $.
+    $( If two elements commute, they commute with each others inverses.  This
+       theorem is the specific case of the first element commuting with the
+       inverse of the second element, but note that since the last hypothesis
+       is symmetric, we can interchange ` X ` and ` Y ` in the result.
+       (Contributed by SN, 29-Jan-2025.) $)
+    grpcominv1 $p |- ( ph -> ( X .+ ( N ` Y ) ) = ( ( N ` Y ) .+ X ) ) $=
+      ( cfv co wceq grpassd 3eqtr4rd wcel grpinvcld eqid grplinvd oveq1d eqtr2d
+      c0g sn-grplidd oveq2d grpasscan2d cgrp wb grpcld grprcan syl13anc mpbid )
+      AFGEOZCPZGCPZUPFCPZGCPZQZUQUSQZAUPFGCPZCPZFUTURAUPGCPZFCPZUPGFCPZCPFVDABC
+      DUPGFHIKABDEGHJKMUAZMLRAVFDUFOZFCPFAVEVIFCABCDEGVIHIVIUBZJKMUCUDABCDFVIHI
+      VJKLUGUEAVCVGUPCNUHSABCDUPFGHIKVHLMRABCDEFGHIJKLMUISADUJTUQBTUSBTGBTVAVBU
+      KKABCDFUPHIKLVHULABCDUPFHIKVHLULMBCDUQUSGHIUMUNUO $.
+  $}
+
+  ${
+    odm1inv.x $e |- X = ( Base ` G ) $.
+    odm1inv.o $e |- O = ( od ` G ) $.
+    odm1inv.t $e |- .x. = ( .g ` G ) $.
+    odm1inv.i $e |- I = ( invg ` G ) $.
+    odm1inv.g $e |- ( ph -> G e. Grp ) $.
+    odm1inv.1 $e |- ( ph -> A e. X ) $.
+    $( The (order-1)th multiple of an element is its inverse.  (Contributed by
+       SN, 31-Jan-2025.) $)
+    odm1inv $p |- ( ph -> ( ( ( O ` A ) - 1 ) .x. A ) = ( I ` A ) ) $=
+      ( cfv co c1 wcel wceq eqid syl csg c0g cmin odid mulg1 oveq12d cgrp odcld
+      cz nn0zd 1zzd mulgsubdir syl13anc grpinvval2 syl2anc 3eqtr4d ) ABFNZBCOZP
+      BCOZDUANZOZDUBNZBUTOZUQPUCOBCOZBENZAURVBUSBUTABGQZURVBRMBCDFGVBHIJVBSZUDT
+      AVFUSBRMGCDBHJUETUFADUGQZUQUIQPUIQVFVDVARLAUQABGDFHIMUHUJAUKMGCDUQUTPBHJU
+      TSZULUMAVHVFVEVCRLMGDUTEBVBHVIKVGUNUOUP $.
+  $}
+
+  ${
+    $d S a $.  $d G a $.  $d ph a $.
+    finodsubmsubg.o $e |- O = ( od ` G ) $.
+    finodsubmsubg.g $e |- ( ph -> G e. Grp ) $.
+    finodsubmsubg.s $e |- ( ph -> S e. ( SubMnd ` G ) ) $.
+    finodsubmsubg.1 $e |- ( ph -> A. a e. S ( O ` a ) e. NN ) $.
```

Comment:
Maybe worth it to generalize to
```
    finodsubmsubg.1 $e |- ( ph -> A. a e. S ( O ` a ) e. NN0 ) $.
```
?

Reply:
the order 0 represents the infinite case where no multiple of the element is the identity, so I don't think the generalization is possible

Reply:
Ok... (???) !!!  In standard terminology, no element has order zero.  I just looked at https://us.metamath.org/mpeuni/df-od.html and it's indeed a pretty weird choice to define as zero the "order" of an element of infinite order. That definition df-od is also unnatural in its use of a special case (the "if" operator).